client/data/reader/use-login-window.ts

```diff
@@ -0,0 +1,68 @@
+import config from '@automattic/calypso-config';
+import { createAccountUrl } from 'calypso/lib/paths';
+
+interface UseLoginWindowProps {
+	onLoginSuccess: () => void;
+}
+
+interface UseLoginWindowReturn {
+	login: () => void;
+	createAccount: () => void;
+}
+
+export default function useLoginWindow( {
+	onLoginSuccess,
+}: UseLoginWindowProps ): UseLoginWindowReturn {
+	const isBrowser: boolean = typeof window !== 'undefined';
+	const environment: string = config( 'env_id' );
+	let domain: string = 'wordpress.com';
```

Comment:
Im seeing linting errors in my VSCode around a lot of these, basically saying we don't need to type it since its heavily inferred from the declaration.

<img width="825" alt="Screenshot 2023-10-17 at 1 17 10 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/9147a4b9-5e9b-41e0-ba01-4ade1ef3082f">

Reply:
Tided this up -wasn't showing up in phpstorm for some reason.